devicetypes/smartthings/zigbee-lock.src/zigbee-lock.groovy

```diff
@@ -1158,8 +1158,9 @@ def reportsBatteryIncorrectly() {
 			"YRD210 PB DB",
 			"YRD220/240 TSDB",
 			"YRL210 PB LL",
+			"c700000202" //YDF40
 	]
-	return (isYaleLock() && device.getDataValue("model") in badModels)
+	return device.getDataValue("model") in badModels
```

Comment:
@greens Shouldn't this part of code be refactored a little bit too ?
https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/5236452a7ace5658d377bd4bc0c867be37ea1ac2/devicetypes/smartthings/zigbee-lock.src/zigbee-lock.groovy#L481

Currently this value is redundantly set twice in case of Yale locks.
That part of the code might look like -> 
```
// Handling Yale locks incorrect battery reporting issue
if (reportsBatteryIncorrectly()) {
	responseMap.value = Integer.parseInt(descMap.value, 16)
} else {
	responseMap.value = Math.round(Integer.parseInt(descMap.value, 16) / 2)
}
```